File: crawler/backend/internal/domain/repository/site_repository.go
package repository

import (
	"context"
	"github.com/startup-job-board/crawler/internal/domain/entity"
)

// SiteRepository defines the interface for crawl site persistence
type SiteRepository interface {
	Create(ctx context.Context, site *entity.CrawlSite) error
	Update(ctx context.Context, site *entity.CrawlSite) error
	Delete(ctx context.Context, id string) error
	FindByID(ctx context.Context, id string) (*entity.CrawlSite, error)
	FindAll(ctx context.Context) ([]*entity.CrawlSite, error)
	FindActive(ctx context.Context) ([]*entity.CrawlSite, error)
	UpdateLastCrawledAt(ctx context.Context, id string) error
}

File: crawler/frontend/src/domain/repositories/SiteRepository.ts
import { CrawlSite } from '../entities/CrawlSite';
import { CreateSiteInput } from '../value-objects/CreateSiteInput';
import { UpdateSiteInput } from '../value-objects/UpdateSiteInput';

export interface SiteRepository {
  findAll(): Promise<CrawlSite[]>;
  findById(id: string): Promise<CrawlSite>;
  create(input: CreateSiteInput): Promise<CrawlSite>;
  update(id: string, input: UpdateSiteInput): Promise<CrawlSite>;
  delete(id: string): Promise<void>;
}

